### ISSUE TITLE
failed to use fcmaes on Debian 10

### ISSUE THREAD

**Issue by juliendehos** [dietmarwo/fast-cma-es#2]
Hi, I'm trying to use fcmaes on Debian stable (10) but I get an error `GLIBCXX_3.4.26 not found`.

I guess `libacmalib.so` needs a more recent GCC but my GCC is not very old (8.3). Is it possible to have a more portable pre-built `.so` file, or its source code ?

Here is what I've done:

```
$ sudo apt install intel-mkl

$ virtualenv env -p python3

$ source env/bin/activate

(env) $ pip install fcmaes

(env) $ python
Python 3.7.3 (default, Dec 20 2019, 18:57:59)
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.

>>> from fcmaes import cmaescpp
Traceback (most recent call last):
File "<stdin>", line 1, in <module>
File "/home/toto/test-fcmaes/env/lib/python3.7/site-packages/fcmaes/cmaescpp.py", line 129, in <module>
libcmalib = ct.cdll.LoadLibrary(basepath + '/lib/libacmalib.so')
File "/usr/lib/python3.7/ctypes/__init__.py", line 434, in LoadLibrary
return self._dlltype(name)
File "/usr/lib/python3.7/ctypes/__init__.py", line 356, in __init__
self._handle = _dlopen(self._name, mode)
OSError: /lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by /home/toto/test-fcmaes/env/lib/python3.7/site-packages/fcmaes/lib/libacmalib.so)

```

**Comment by dietmarwo** [comment]
Hi Julien,
did you see https://github.com/dietmarwo/fast-cma-es/tree/master/_fcmaescpp
 ?
there is a cmake file you can adapt and compile at your site.
You need also Armadillo. If you have success, can you please send me your
cmake file so that I can see which compiler works for Debian?
I plan to add an alternative .so file to github, so that others can switch
if necessary.

Dietmar Wolz

Am Mi., 4. März 2020 um 16:16 Uhr schrieb Julien Dehos <
notifications@github.com>:

**Comment by dietmarwo** [comment]
Hi Julien,
uploaded a version I built with gcc 7 to pip,
so you could try a
pip install fcmaes --upgrade
Hope gcc7 is old enough.

Dietmar

Am Mi., 4. März 2020 um 16:16 Uhr schrieb Julien Dehos <
notifications@github.com>:

**Comment by juliendehos** [comment]
This new version fixes the problem, on my machine. Thank you.

**Comment by timjim333** [comment]
Hi Dietmar,

I had a similar issue trying to `from fcmaes import *` in python.

Sorry if this is a simple question - what does `fcmaes/lib/libgtoplib.so` come from? My CentOS 8 has an even older glic, so I tried to recompile everything in the `_fcmaescpp` directory. I copied my compiled version of  `libacmalib.so` into `fcames/lib` and that seemed to be ok, but a new error came up for `libgtoplib.so`. Is it [this memory application](https://developer.gnome.org/libgtop/stable/)? 

Thank you!
Kind regards,
Tim
